### PR TITLE
Editing/cleanup pass through the Logstash docs.

### PR DESCRIPTION
**Pull request by debadair**
Changes from my review of the Logstash docs. I fixed some bad links, did some light editing, and reorganized the config file topic a bit.

### CODE REVIEW

File: docs/asciidoc/static/configuration.asciidoc [hunk 1]
```diff
@@ -64,74 +52,75 @@ input {
...
 ----------------------------------
 
-The above makes 'path' a 3-element array including all 3 strings.
 [[hash]]
 [float]
 ==== Hash
 
-A hash is basically the same syntax as Ruby hashes.
-The key and value are simply pairs, such as:
+A hash is a collection of key value pairs. Hashes are specified using the Ruby hash
```
Comment (jordansissel) [inline docs/asciidoc/static/configuration.asciidoc]: +1 on this improvement. The old version and this one are slightly inaccurate, I dont' know if we say elsewhere, but hashes are "ruby hash" syntax without commas.

File: docs/asciidoc/static/configuration.asciidoc [hunk 2]
```diff
@@ -258,14 +271,14 @@ output {
 [float]
 === Conditionals
 
-Sometimes you only want a filter or output to process an event under
-certain conditions. For that, you'll want to use a conditional!
+Sometimes you only want a plugin to process an event under
```
Comment (jordansissel) [inline docs/asciidoc/static/configuration.asciidoc]: In this case, I think 'filter or output' is more accurate. Conditionals are only valid for these plugin types and are not used in inputs or codecs.

Thoughts?

Comment (jordansissel) [inline docs/asciidoc/static/configuration.asciidoc]: Maybe "When filtering or outputting events, you may want to make decisions about how to act on certain events. For that, you can use a conditional" ?

File: docs/asciidoc/static/configuration.asciidoc [hunk 3]
```diff
@@ -350,7 +362,7 @@ output {
 }
 ----------------------------------
 
-Here are some examples for testing with the in conditional:
+Here are some examples for testing with the `in` conditional:
```
Comment (jordansissel) [inline docs/asciidoc/static/configuration.asciidoc]: good catch!

File: docs/asciidoc/static/configuration.asciidoc (hunk 2 above)
Comment (debadair) [inline docs/asciidoc/static/configuration.asciidoc]: I agree, being more specific is better in this case.

File: docs/asciidoc/static/getting-started-with-logstash.asciidoc [hunk 4]
```diff
@@ -225,18 +236,18 @@ output {
...
 bin/logstash -f logstash-simple.conf
 ----------------------------------
 
-Et voilà! Logstash will read in the configuration file you just created and run as in the example we saw earlier. Note that we used the '-f' to read in the file, rather than the '-e' to read the configuration from the command line. This is a very simple case, of course, so let's move on to some more complex examples.
+Et voilà! Logstash reads  the specified configuration file and outputs to both Elasticsearch and stdout. This is a very simple case, of course, so let's move on to some more complex examples.
 
 [float]
 ==== Filters
-Filters are an in-line processing mechanism which provide the flexibility to slice and dice your data to fit your needs. Let's see one in action, namely the *grok filter*.
+Filters are an in-line processing mechanism that provide the flexibility to slice and dice your data to fit your needs. Let's take a look at somes filter in action. The following configuration file sets up the *grok* and *date* filters.
```
Comment (jordansissel) [inline docs/asciidoc/static/getting-started-with-logstash.asciidoc]: > a look at the somes filter

The somes filter?

Comment (debadair) [inline docs/asciidoc/static/getting-started-with-logstash.asciidoc]: Now that's a nice typo. Good catch. (Should have been "some filters".)

File: docs/asciidoc/static/configuration.asciidoc [hunk 5]
```diff
@@ -64,74 +52,75 @@ input {
 }
 ----------------------------------
 
-The above configures two file separate inputs. Both set two configuration settings each: 'path' and 'type'. Each plugin has different settings for configuring it; seek the documentation for your plugin to learn what settings are available and what they mean. For example, the [file input][fileinput] documentation will explain the meanings of the path and type settings.
+In this example, two settings are configured for each of the file inputs: 'path' and 'type'. 
 
-[float]
-=== Value Types
+The settings you can configure vary according to the plugin type. For information about each plugin, see http://www.elasticsearch.org/guide/en/logstash/current/input-plugins.html[Input Plugins], http://www.elasticsearch.org/guide/en/logstash/current/output-plugins.html[Output Plugins], http://www.elasticsearch.org/guide/en/logstash/current/filter-plugins.html][Filter Plugins], and http://www.elasticsearch.org/guide/en/logstash/current/codec-plugins.html][Codec Plugins].
```
Comment (suyograo) [inline docs/asciidoc/static/configuration.asciidoc]: can we use relative links `<<input-plugins, Input Plugins>>`

Comment (debadair) [inline docs/asciidoc/static/configuration.asciidoc]: Yes, they should be relative links.

File: docs/asciidoc/static/getting-started-with-logstash.asciidoc [hunk 6]
```diff
@@ -143,78 +145,87 @@ Congratulations! You've successfully stashed logs in Elasticsearch via Logstash.
...
-* *mutate*: The mutate filter allows you to do general mutations to fields. You can rename, remove, replace, and modify fields in your events.
+* *grok*: parse and structure arbitrary text. Grok is currently the best way in Logstash to parse unstructured log data into something structured and queryable. With 120 patterns built-in to Logstash, it's more than likely you'll find one that meets your needs!
+* *mutate*: perform general transformations on event fields. You can rename, remove, replace, and modify fields in your events.
 * *drop*: drop an event completely, for example, 'debug' events.
 * *clone*: make a copy of an event, possibly adding or removing fields.
-* *geoip*: adds information about geographical location of IP addresses (and displays amazing charts in kibana)
+* *geoip*: add information about geographical location of IP addresses (also displays amazing charts in Kibana!)
+
+For more information about the available filters, see http://www.elasticsearch.org/guide/en/logstash/current/filter-plugins.html[filter plugins].
```
Comment (suyograo) [inline docs/asciidoc/static/getting-started-with-logstash.asciidoc]: Relative link? <<filter-plugins, filter plugins>>

**Comment by jordansissel** [comment]
Other than my few comments, this looks great! 

@debadair thanks so much for working on this :)

**Comment by debadair** [comment]
Any ideas what I might have done to cause this error:

using GIT_SSH to set credentials private key to use with github
 > /usr/bin/git fetch --tags --progress git@github.com:elasticsearch/logstash.git +refs/pull/*:refs/remotes/origin/pr/*
ERROR: Error fetching remote repo 'origin'
ERROR: Error fetching remote repo 'origin'

**Comment by jsvd** [comment]
it has been occurring randomly, has nothing to do with your changes

**Comment by jsvd** [comment]
test this please

File: docs/asciidoc/static/getting-started-with-logstash.asciidoc [hunk 7]
```diff
@@ -1,14 +1,12 @@
...
-Using Elasticsearch as a backend datastore, and kibana as a frontend reporting tool, Logstash acts as the workhorse, creating a powerful pipeline for storing, querying and analyzing your logs. With an arsenal of built-in inputs, filters, codecs and outputs, you can harness some powerful functionality with a small amount of effort. So, let's get started!
+Logstash provides a powerful pipeline for storing, querying, and analyzing your logs. When using Elasticsearch as a backend data store and Kibana as a frontend reporting tool, Logstash acts as the workhorse.  It includes an arsenal of built-in inputs, filters, codecs, and outputs, enabling you to harness some powerful functionality with a small amount of effort. So, let's get started!
 
 [float]
 ==== Prerequisite: Java
-The only prerequisite required by Logstash is a Java runtime. You can check that you have it installed by running the  command `java -version` in your shell. Here's something similar to what you might see:
+The only prerequisite for installing Logstash is a Java runtime. You can verify that you have it installed by running the  command `java -version` in your shell. Here's something similar to what you might see:
```
Comment (kurtado) [inline docs/asciidoc/static/getting-started-with-logstash.asciidoc]: Technically, the prequisite isn't for *installing*, but *running* Logstash :)

Comment (debadair) [inline docs/asciidoc/static/getting-started-with-logstash.asciidoc]: True. Do we want to be more specific about what the minimum version of Java is? The ES guide is similarly vague,  but the reference explictly calls out Java 7 as the minimum version. http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/setup.html#jvm-version

Does it make sense to be consistent about the versions we are recommending?

Comment (suyograo) [inline docs/asciidoc/static/getting-started-with-logstash.asciidoc]: +1 on mentioning Java 1.7 as min version.

**Comment by suyograo** [comment]
LGTM

**Comment by jordansissel** [comment]
LGTM

**Comment by electrical** [comment]
According to the bot its only been merged into 1.5 branch. any reason why its not in master branch as well?

**Comment by purbon** [comment]
Good catch, should be also in master :+1:

**Comment by suyograo** [comment]
@electrical @purbon it was merged manually to master and we used the bot to merge to 1.5

**Comment by jordansissel** [comment]
Yeah, the green merge button was used to shim it into master.

**Pull request merged**